incubator/hnc/pkg/reconcilers/hierarchical_namespace.go

```diff
@@ -0,0 +1,49 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconcilers
+
+import (
+	"github.com/go-logr/logr"
+	api "github.com/kubernetes-sigs/multi-tenancy/incubator/hnc/api/v1alpha1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// HierarchicalNamespaceReconciler reconciles HierarchicalNamespace CRs to make sure
+// all the hierarchical namespaces are properly maintained.
+type HierarchicalNamespaceReconciler struct {
```

Comment:
I'm thinking it's a bit confusing to have a `HierarchicalNamespace` and `Hierarchy` reconciler. Maybe the other one should switch to `HierarchyConfig`?

Reply:
sg